jenkins/cli.sls

```diff
@@ -1,61 +1,37 @@
 {% from "jenkins/map.jinja" import jenkins with context %}
+{% import "jenkins/macros/cli_macro.jinja" as cli_macro %}
 
-{%- macro fmtarg(prefix, value)-%}
-{{ (prefix + ' ' + value) if value else '' }}
-{%- endmacro -%}
-{%- macro jenkins_cli(cmd) -%}
-{{ ' '.join(['java', '-jar', jenkins.cli_path, '-s', jenkins.master_url, fmtarg('-i', jenkins.get('privkey')), cmd]) }} {{ ' '.join(varargs) }}
-{%- endmacro -%}
+{% if grains['os_family'] == 'RedHat' %}
+  {% set listening_tool = "curl" %}
+{% else %}
+  {% set listening_tool = jenkins.netcat_pkg %}
+{% endif %}
 
-{% set timeout = 360 %}
-
-jenkins_listening:
-  pkg.installed:
-    - name: {{ jenkins.netcat_pkg }}
-  cmd.wait:
-    - name: "until nc -z localhost {{ jenkins.jenkins_port }}; do sleep 1; done"
-    - timeout: 10
-    - require:
-      - service: jenkins
-    - watch:
-      - service: jenkins
-
-jenkins_serving:
-  pkg.installed:
-    - name: curl
+check_if_jenkins_server_runs:
+  cmd.run:
```

Comment:
This command would fail if the package is not installed. Is there a good reason to replace ``cmd.wait`` and with ``cmd.run`` since ``cmd.run`` will always run whenever the state is invoked.